hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/streams/assertions/AbstractEventualStreamAssertion.java

```diff
@@ -82,11 +82,18 @@ public void assertHasPassed() {
             final var eventualResult = result.get();
             unsubscribe();
             if (!eventualResult.passed()) {
-                Assertions.fail(eventualResult.getErrorDetails());
+                Assertions.fail(assertionDescription() + " ended with result: " + eventualResult.getErrorDetails());
             }
         } catch (final InterruptedException e) {
+            unsubscribe();
```

Comment:
Given L83 exists too - does it matter exactly _when_ `unsubscribe` is called?  If not, could be done in a `finally`.

Reply:
Moved to `finally`.